### ISSUE TITLE
Bug: parser breaks on folders that has been excluded

### ISSUE THREAD

**Issue by MichailShcherbakov** [typescript-eslint/typescript-eslint#8798]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```TypeScript
- Create the "dist" folder with ".ts" files;
- Run command: "npx eslint .".
```

### ESLint Config

```javascript
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";

export default tseslint.config(eslint.configs.recommended, ...tseslint.configs.recommended, {
  ignores: ["node_modules/*", "dist/*", "coverage/*"],
  languageOptions: {
    ecmaVersion: 2020,
    sourceType: "module",
    parserOptions: {
      project: true,
    },
  },
});
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "module": "ESNext",
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "skipLibCheck": true,
    "moduleResolution": "Bundler",
    "allowImportingTsExtensions": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "strict": true,
    "alwaysStrict": true,
    "strictNullChecks": true,
    "noImplicitAny": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,
    "types": ["vitest/globals"]
  },
  "include": ["**/*.ts", "**/*.js"],
  "exclude": ["node_modules", "dist", "coverage"]
}
```

### Expected Result

It should not parse files in folders that has been excluded.

### Actual Result

It parses files in folders that has been excluded.

```
0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/dist/types/utils/is_empty.d.ts` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `eslint`     | `8.57.0` |
| `typescript-eslint`            | `7.4.0` |
| `typescript`            | `5.4.3` |
| `node`            | `20.11.1` |

**Comment by bradzacher** [comment]
See this comment which should help understand where you went wrong 
https://github.com/typescript-eslint/typescript-eslint/issues/8787#issuecomment-2024649691

**Comment by MichailShcherbakov** [comment]
@bradzacher yeah, you're right. It should be used without any other keys in the configuration object and then it works as expected.
